pkg/routeagent_driver/handlers/healthchecker/healthchecker.go

```diff
@@ -0,0 +1,331 @@
+			return nil
+		}
+
+		logger.Infof("HealthChecker is already running for %q - stopping", endpoint.Name)
+		pingerObject.Stop()
+		delete(h.pingers, endpoint.Spec.CableName)
+	}
+
+	pingerConfig := pinger.Config{
+		IP:                 endpoint.Spec.HealthCheckIP,
+		MaxPacketLossCount: h.config.MaxPacketLossCount,
+	}
+
+	if h.config.PingInterval != 0 {
+		pingerConfig.Interval = time.Second * time.Duration(h.config.PingInterval)
+	}
+
+	if h.config.MaxPacketLossCount != 0 {
+		pingerConfig.MaxPacketLossCount = h.config.MaxPacketLossCount
+	}
+
+	newPingerFunc := h.config.NewPinger
+	if newPingerFunc == nil {
+		newPingerFunc = pinger.NewPinger
+	}
+
+	pingerObject := newPingerFunc(pingerConfig)
+	h.pingers[endpoint.Spec.CableName] = pingerObject
+	pingerObject.Start()
+
+	logger.Infof("CableEngine HealthChecker started pinger for CableName: %q with HealthCheckIP %q",
+		endpoint.Spec.CableName, endpoint.Spec.HealthCheckIP)
+
+	return nil
+}
+
+func (h *controller) RemoteEndpointDeleted(endpoint *submarinerv1.Endpoint) error {
+	h.Lock()
+	defer h.Unlock()
+
+	if pingerObject, found := h.pingers[endpoint.Spec.CableName]; found {
+		pingerObject.Stop()
+		delete(h.pingers, endpoint.Spec.CableName)
+	}
+
+	delete(h.remoteEndpoint, endpoint.Name)
+
+	return nil
+}
+
+func (h *controller) Init() error {
+	go func() {
+		wait.Until(h.syncRouteAgentStatus, RouteAgentUpdateInterval, h.stopCh)
+	}()
+
+	return nil
+}
+
+// TransitionToNonGateway is called once for each transition of the local node from Gateway to a non-Gateway.
+func (h *controller) TransitionToNonGateway() error {
+	if h.config.HealthCheckerEnabled {
+		for i := range h.remoteEndpoint {
+			err := h.processEndpointCreatedOrUpdated(h.remoteEndpoint[i])
+			if err != nil {
+				logger.Warningf("Error processing remote endpoint %s: %v", h.remoteEndpoint[i].Name, err)
+			}
+		}
+	}
+
+	return nil
+}
+
+// TransitionToGateway is called once for each transition of the local node from non-Gateway to a Gateway.
+func (h *controller) TransitionToGateway() error {
+	if h.config.HealthCheckerEnabled {
+		for i := range h.pingers {
+			h.pingers[i].Stop()
+			delete(h.pingers, i)
+		}
+	}
+
+	close(h.stopCh)
+
+	err := h.client.Delete(context.TODO(), h.localNodeName, metav1.DeleteOptions{})
+	if err != nil {
+		logger.Warningf("Error deleting RouteAgent: %s: %v", h.localNodeName, err)
+	}
+
+	return nil
+}
+
+func (h *controller) GetNetworkPlugins() []string {
+	return []string{event.AnyNetworkPlugin}
+}
+
+func (h *controller) GetName() string {
+	return "routeAgent-health-checker"
+}
+
+func (h *controller) syncRouteAgentStatus() {
+	routeAgent := h.generateRouteAgentObject()
+	routeAgent.Status.RemoteEndpoints = []submarinerv1.RemoteEndpoint{}
+
+	for _, endpoint := range h.remoteEndpoint {
+		pingerObject, found := h.pingers[endpoint.Spec.CableName]
+		if !found {
+			logger.Warningf("Pinger not found for %q", endpoint.Spec.CableName)
+			continue
+		}
```

Comment:
A pinger will be missing if health check is not enabled or we're on the gateway. We should still create a `RemoteEndpoint` entry and set the `StatusMessage`, ie if `!HealthCheckerEnabled` then "_Health check is not enabled_" or if `IsOnGateway` then "_Health check is not performed on gateway nodes"_.  We could leave the `Status` field empty but it's defined as an enum. None of the current  enums apply so we can add `ConnectionNone`.